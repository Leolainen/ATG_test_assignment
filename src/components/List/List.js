import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";
import styles from "./styles.module.css";

const List = (props) => {
  const { className, children, component: Component = "ul", ...other } = props;

  return (
    <Component className={classnames(styles.root, className)} {...other}>
      {children}
    </Component>
  );
};

List.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

List.uiName = "List";

export default List;
