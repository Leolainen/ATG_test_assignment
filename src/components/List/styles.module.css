.root {
  position: relative;
  list-style: none;
  margin: 0;
  padding: 8px 0;
}
