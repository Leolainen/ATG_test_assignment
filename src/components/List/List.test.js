import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import List from "./List";

describe("<List />", () => {
  afterEach(cleanup);

  describe("Snapshots", () => {
    it("should render", () => {
      const { asFragment } = render(<List>list</List>);

      expect(asFragment()).toMatchSnapshot();
    });
  });
});
