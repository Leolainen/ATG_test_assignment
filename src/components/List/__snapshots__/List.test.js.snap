// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> Snapshots should render 1`] = `
<DocumentFragment>
  <ul
    class="root"
  >
    list
  </ul>
</DocumentFragment>
`;
